proposals/nnnn-PolicyModuleDesign.md

```diff
@@ -0,0 +1,100 @@
+# Policy Module Design
+
+* Proposal: [SDL-NNNN](nnnn-PolicyModuleDesign.md)
+* Author: [Elisey Zamakhov](https://github.com/pestOO)
+* Status: **Awaiting review**
+* Impacted Platforms: Core
+
+## Introduction
+
+At this moment SDL Core has a list of SDL Policies with familiar requirements and different code base:
+
+* [OpenSource (O-S) Policy](https://github.com/smartdevicelink/sdl_core/tree/master/src/components/policy) - Opensource Policy functionality
+* [Ford-Specific (F-S) Policy ](https://github.com/CustomSDL/sdl_panasonic/tree/develop/src/components/policy) - Upgraded and enchanted Policy for SYNC3 integration
+* *new* OpenSource External Proprietary (O-EP) - will be available in OpenSDL Core as a port of F-S Policy to open-source
+* [RemoteControl (RC) Policy](https://github.com/CustomSDL/sdl_core/tree/master/src/components/policy) - extension for RemoteControl functionality
+
+This proposal suggest a way to merge all (or at least most of) this Policies to one base source with a list of specific extensions.
+
+*Note:* For further details about Policy component and dependencies, please refer to [SAD - Component View](https://smartdevicelink.com/en/guides/core/software-architecture-document/components-view/).
+
+## Motivation
+
+Current approach with handling O-S, O-EP and RC- Policies as a separate source bases leads to double and triple design, implementation and bug-fixing for each code base.
+Each bug-fixes, enchantments and improvements need to be applied to each source code: O-S, F-S, O-EP, RC.
+
+## Proposed solution
+
+SDL Core Policy functionality could be split to the following separate components with own area of responsibility
+
+* **Base Policy**
+* **Open Policy extension**
+* **OpenSource External Proprietary Policy extension** (F-S ported to open-source)
+* **Remote Control Policy extension**
+
+![Policy Module Design](assets/PolicyModuleDesign.png)
+
+* **Base Policy** provides list of interfaces for easy customer-specific integration and base Policy functionality:
+  * Application behavior (RPC, heartbeat, hmi-types) management, authentication and revocation
+  * Mobile RPC Permission management
+  * Multiple persistent storage (DB, file, RAM)
+  * Cloud PT Update, Retry seq
+  * Preloaded PT 
+  * Statistics
+* **Extension** is a set of interfaces for each Policy Base feature add-on
+* **Open Policy extension** - extends base with O-S only functionality:
+  * SQlite Database
```

Comment:
@pestOO 
I think this is base functionality since its being used now as primary approach to store policy data for every implementation.